syntax = "proto3";

package streaming;

service Streamer {
  rpc ProcessTextStream(stream TextRequest) returns (TextResponse);
}

message TextRequest {
  string message = 1;
  int32 channel_id = 2;
  string client_id = 3;
}

message TextResponse {
  int32 message_count = 1;
}